src/Base/index.js

```diff
@@ -1,78 +1,77 @@
 import React from 'react';
-import {ActivityIndicator, Text, View, Pressable} from 'react-native';
+import {Text, View, TextInput} from 'react-native';
 
-const Button = ({
-  disabled = false,
-  onPress,
-  busy = false,
-  loaderIcon = <ActivityIndicator color="#fff" />,
-  busyText = 'Processing..',
+const Input = ({
+  onChangeText,
+  iconPosition,
+  icon,
+  style,
+  value,
+  placeholder = 'Enter Text',
+  label = null,
+  error,
   ...props
 }) => {
-  const textStyles = {
-    ...(props.textStyles || {
-      fontWeight: 'bold',
-      color: props.textColor || '#fff',
-      letterSpacing: 0,
-    }),
+  const [focused, setFocused] = React.useState(false);
+
+  const getFlexDirection = () => {
+    if (icon && iconPosition) {
+      if (iconPosition === 'left') {
+        return 'row';
+      } else if (iconPosition === 'right') {
+        return 'row-reverse';
+      }
+    }
+  };
+
+  const getBorderColor = () => {
+    if (error) {
+      return '#DC2626';
+    }
+
+    if (focused) {
+      return '#E4E4E7';
+    }
   };
 
   return (
-    <Pressable
-      disabledd={disabled}
-      onPress={() => {
-        if (!disabled) {
-          onPress();
-        }
-      }}
-      style={({pressed}) => [
-        {
-          backgroundColor: pressed
-            ? props.onPressColor || '#27272A'
-            : props.backgroundColor || '#18181B',
-          width: '100%',
-          borderColor: props.borderColor || '#27272A',
+    <View style={{paddingVertical: 12}}>
+      {label && <Text>{label}</Text>}
+
+      <View
+        style={{
+          height: 42,
```

Comment:
@Dhruv-Rathi User should be able to override these styles, passing them as props, that's why in button component i used something like:

```
            ...(props.loaderIconWrapperStyles || {
              alignItems: 'center',
              flexDirection: 'row',
              justifyContent: 'center',
            }),
```